pvlib/iotools/solcast.py

```diff
@@ -0,0 +1,443 @@
+    # dewpoint_temp -> temp_dew (deg C)
+    ParameterMap("dewpoint_temp", "temp_dew"),
+    # gti (W/m^2) -> poa_global (W/m^2)
+    ParameterMap("gti", "poa_global"),
+    # wind_speed_10m (m/s) -> wind_speed (m/s)
+    ParameterMap("wind_speed_10m", "wind_speed"),
+    # wind_direction_10m (deg) -> wind_direction  (deg)
+    ParameterMap("wind_direction_10m", "wind_direction"),
+    # azimuth -> solar_azimuth (degrees) (different convention)
+    ParameterMap(
+        "azimuth", "solar_azimuth", lambda x: abs(x) if x <= 0 else 360 - x
+    ),
+    # precipitable_water (kg/m2) -> precipitable_water (cm)
+    ParameterMap("precipitable_water", "precipitable_water", lambda x: x*10),
+    # zenith -> solar_zenith
+    ParameterMap("zenith", "solar_zenith")
+]
+
+
+def get_solcast_tmy(
+    latitude, longitude, api_key, map_variables=True, **kwargs
+):
+    """Get the irradiance and weather for a
+    Typical Meteorological Year (TMY) at a requested location.
+
+    Derived from satellite (clouds and irradiance over
+    non-polar continental areas) and numerical weather models (other data).
+    The TMY is calculated with data from 2007 to 2023.
+
+    Parameters
+    ----------
+    latitude : float
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float
+        in decimal degrees, between -180 and 180, east is positive
+    api_key : str
+        To access Solcast data you will need an API key [1]_.
+    map_variables: bool, default: True
+        When true, renames columns of the DataFrame to pvlib variable names
+        where applicable. See variable :const:`VARIABLE_MAP`.
+    kwargs:
+        Optional parameters passed to the API.
+        See [2]_ for full list of parameters.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        containing the values for the parameters requested.The times
+        in the DataFrame index indicate the midpoint of each interval.
+
+    Examples
+    --------
+    >>> pvlib.iotools.solcast.get_solcast_tmy(
+    >>>      latitude=-33.856784,
+    >>>      longitude=151.215297,
+    >>>      api_key="your-key"
+    >>> )
+
+    you can pass any of the parameters listed in the API docs,
+    like ``time_zone``. Here we set the value of 10 for
+    "10 hours ahead of UTC":
+
+    >>> pvlib.iotools.solcast.get_solcast_tmy(
+    >>>     latitude=-33.856784,
+    >>>     longitude=151.215297,
+    >>>     time_zone=10,
+    >>>     api_key="your-key"
+    >>> )
+
+    References
+    ----------
+    .. [1] `Get an API Key <https://toolkit.solcast.com.au/register>`_
+    .. [2] `Solcast API Docs <https://docs.solcast.com.au/>`_
+    """
+
+    params = dict(
+        latitude=latitude,
+        longitude=longitude,
+        format="json",
+        **kwargs
+    )
+
+    data = _get_solcast(
+        endpoint="tmy/radiation_and_weather",
+        params=params,
+        api_key=api_key,
+        map_variables=map_variables
+    )
+
+    return data, {}

```

Comment:
Is there no way to get metadata (data version, e.g.) back from the API?

Let's at least echo the `latitude` and `longitude` inputs in the metadata dict.

Reply:
data version not yet! I have added the lat/lon as suggested.